### PR TITLE
Use Request objects and controller validation for campaign endpoints.

### PR DESCRIPTION
**Pull request by DFurnes**
# Changes
 - Use Laravel 5 [Request](http://laravel.com/docs/master/requests#accessing-the-request) objects & [Controller validation](http://laravel.com/docs/master/validation#a-complete-example) for campaign endpoints.
 - Cleans up campaign controller using Laravel 5 style response helpers.
 - Use exceptions for error handling, and set handler to render with standard JSON format if `Accept: application/json` header is set. References #84.

For review: @angaither @jonuy

### CODE REVIEW

File: app/Http/Controllers/CampaignController.php [hunk 1]
```diff
@@ -52,24 +52,16 @@ public function show($term, $id)
...
-
-        // Validate request
-        $validator = Validator::make($request, [
-            'campaign_id' => ['required', 'integer'],
+        // Validate request body
+        $this->validate($request, [
             'source' => ['required']
         ]);
 
-        if ($validator->fails()) {
```
Comment (jonuy) [inline app/Http/Controllers/CampaignController.php]: If validation does fail, what response does the client get back now?

Comment (DFurnes) [inline app/Http/Controllers/CampaignController.php]: It'll throw an exception and return `422 Unprocessable Entity` with the validation errors.

Comment (jonuy) [inline app/Http/Controllers/CampaignController.php]: Ah an exception breaks it out of the control flow. Ok :+1:

File: app/Exceptions/Handler.php [hunk 2]
```diff
@@ -37,6 +37,32 @@ public function report(Exception $e)
...
     {
+        // If client requests it, render exception as JSON object
+        if ($request->ajax() || $request->wantsJson()) {
+            $code = 500;
+
+            if($this->isHttpException($e)) {
+                $code = $e->getStatusCode();
+            }
+
+            $response = [
```
Comment (jonuy) [inline app/Exceptions/Handler.php]: nice

**Comment by DFurnes** [comment]
@jonuy Is this safe to merge?

File: app/Http/Controllers/CampaignController.php [hunk 3]
```diff
@@ -107,54 +102,48 @@ public function signup($campaign_id)
...
         // Create a reportback via the Drupal API, and store reportback ID in Northstar
         $reportback_id = $this->drupal->campaignReportback($user->drupal_id, $campaign_id, [
-            'quantity' => $request['quantity'],
-            'why_participated' => $request['why_participated'],
-            'file' => $request['file'],
-            'caption' => $request['caption']
+            'quantity' => $request->input('quantity'),
+            'why_participated' => $request->input('why_participated'),
+            'file' => $request->input('file'),
+            'caption' => $request->input('caption')
```
Comment (angaither) [inline app/Http/Controllers/CampaignController.php]: can you `$this->fill($input);` ??

Comment (DFurnes) [inline app/Http/Controllers/CampaignController.php]: Mmm, good call. :+1:

**Comment by angaither** [comment]
:cool: :cool: :cool:

**Pull request merged**